api/src/pcapi/core/offerers/api.py

```diff
@@ -54,7 +54,7 @@
 
 logger = logging.getLogger(__name__)
 
-UNCHANGED = object()
+UNCHANGED = "UNCHANGED"
```

Comment:
Ca me gêne un peu car si on a un champ qui vaut cette string, on l'a dans l'os. C'est peu probable mais ça montre que c'est pas idéal.

J'invoque l'avis de l'amateur de céleri, @dbaty.

Reply:
Peut-être faire comme dans https://github.com/pass-culture/pass-culture-main/blob/master/api/src/pcapi/core/offers/api.py#L198 ? Càd garder` UNCHANGED = object()`, mais rajouter `None` dans le type possible en entrée dans la méthode.

Reply:
Ma branche de céléri m'a communiqué (par télébathie) son avis : comme Victor, je me dis qu'on devrait garder `object()` par sécurité. Ça impliquerait de typer les arguments avec `str | object`, ce qui est un peu large. :)

Idée de variante (non testée !) : 

```python
class UNCHANGED:
    pass
T_UNCHANGED = typing.Type[UNCHANGED]


def update_offerer(city: str | T_UNCHANGED = UNCHANGED, ...):
```

On ratisserait moins large qu'en typant avec `object`. Et ça reste à peu près lisible. Enfin aussi lisible qu'une annotation de type... ;)

Reply:
Ça marche, effectivement !
Il faut cependant lui préciser `city = typing.cast(str, city)` après le `if _ is not UNCHANGED` vu qu'il ne sait pas que c'est forcément une string après

Reply:
EDIT: contre-proposition pour éviter d'avoir à `cast`, entièrement basée sur [cette discussion](https://github.com/python/typing/issues/236)

```python
class SENTINEL(enum.Enum):
    TOKEN = "SENTINEL"


UNCHANGED = SENTINEL.TOKEN

def update_offerer(
    offerer: models.Offerer,
    city: str | SENTINEL = UNCHANGED,
) -> dict[str, dict[str, str | None]]:
    modified_info = {}

    if city is not UNCHANGED:
        # Ici mypy comprend que city est de type Optional[str]
```